src/msha/middlewares/request.middleware.ts

```diff
@@ -107,8 +107,10 @@ function serveStaticOrProxyReponse(req: http.IncomingMessage, res: http.ServerRe
     logger.silly(` - target: ${chalk.yellow(target)}`);
   }
 
+  const is4xx = res.statusCode >= 400 && res.statusCode < 500;
+
   // if the static app is served by a dev server, forward all requests to it.
-  if (IS_APP_DEV_SERVER()) {
+  if (IS_APP_DEV_SERVER() && !is4xx) {
```

Comment:
I think with this condition, If the user has custom 401/403/404 pages defined (as response overrides), we won't be able to serve them when running in dev server mode.

Reply:
Looks like it needs both checks. Pushed an update that uses dev server if it's a custom url or it's not 4xx.